tests/Doctrine/Tests/ORM/Functional/EntityRepositoryCriteriaTest.php

```diff
@@ -105,6 +105,21 @@ public function testIsNullComparison()
         $this->assertEquals(1, count($dates));
     }
 
+    public function testIsNullComparisonQuery()
+    {
+        $this->loadNullFieldFixtures();
+        $repository = $this->_em->getRepository(DateTimeModel::class);
+
+        $repository->matching(new Criteria(
+            Criteria::expr()->isNull('time')
+        ))->first();
+
+        $this->assertContains(
+            'col_time IS NULL',
+            $this->_sqlLoggerStack->queries[$this->_sqlLoggerStack->currentQuery]['sql']
```

Comment:
Checking for a part of a string in an SQL query is kind of a foggy test compared to the other tests in this class. I would prefer to work and compare with a result.

Reply:
I see your point. But my intention was to demonstrate the query is created as expected given this criteria (due to #7717). What do you think?

Reply:
Despite that the query part was created I like the test to show that `isNull` is also returning the expected result when the query gets executed. Checking the SQL itself isn't the same as having the expected result of the query. This way regressions in the functionality can be covered too for any other unwanted changes that might be introduced besides `'col_time IS NULL'`.